### PR TITLE
chore: define a virtual-platform for all bouncycastle modules in orde…

### PR DESCRIPTION
**Pull request by dhenneke**
…r to get dependency conflicts when only one of the modules is updated

Based on https://docs.gradle.org/current/userguide/dependency_version_alignment.html#sec:virtual_platform

### CODE REVIEW

File: sda-commons-dependency-check/build.gradle [hunk 1]
```diff
@@ -35,3 +39,18 @@ dependencies {
...
+
+/**
+ * A rule that forces all bouncycastle modules to use the same version.
+ */
+class BouncycastleAlignmentRule implements ComponentMetadataRule {
+  @Override
+  void execute(ComponentMetadataContext ctx) {
+    ctx.details.with {
+      if (id.group == "org.bouncycastle") {
+        // declare that the modules belong to the smallrye-config virtual platform
```
Comment (Fox32) [inline sda-commons-dependency-check/build.gradle:51]: I guess I know where you copied that from :wink

Comment (dhenneke) [reply sda-commons-dependency-check/build.gradle]: 😮

**Review by Fox32 — APPROVED** [review]
👍 Fine for me

**Review by christopher-cudennec — APPROVED** [review]
Me too 👍

**Pull request merged**